java-client/src/main/java/co/elastic/clients/elasticsearch/_types/WriteResponseBase.java

```diff
@@ -84,7 +85,7 @@ protected WriteResponseBase(AbstractBuilder<?> builder) {
 
 		this.id = ApiTypeHelper.requireNonNull(builder.id, this, "id");
 		this.index = ApiTypeHelper.requireNonNull(builder.index, this, "index");
-		this.primaryTerm = ApiTypeHelper.requireNonNull(builder.primaryTerm, this, "primaryTerm");
+		this.primaryTerm = builder.primaryTerm;
```

Comment:
For a follow-up PR, `seqNo` should be optional as well. https://github.com/elastic/elasticsearch/blob/a279a302bfe9289e522a4bb32499af1304a70c0b/server/src/main/java/org/elasticsearch/action/DocWriteResponse.java#L291